csvGenerator.ipynb

```diff
@@ -0,0 +1,195 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "6f275807",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import csv\n",
+    "import math\n",
+    "import random\n",
+    "import requests\n",
+    "from CitesphereConnectorTest import *\n",
+    "from CitesphereConnector import CitesphereConnector"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "2c0597b1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "auth_object = authObject()\n",
+    "auth_object.authType  = 'oauth'\n",
+    "auth_object.access_token=\"access-token\"\n",
+    "connector = CitesphereConnector(\"https://diging-dev.asu.edu/citesphere-review/api\", auth_object)\n",
+    "groups=connector.get_groups()\n",
+    "#default max number of items displayed on a collection items page \n",
+    "max_size=50 "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "37167759",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#download files from the collection items\n",
+    "def download_file(folder_path:str,ids:set, citesphere_token:str,):\n",
+    "    path_list = []\n",
+    "    for (file_id, file_name) in list(ids):\n",
+    "        giles_url = f\"https://diging.asu.edu/geco-giles-staging/api/v2/resources/files/{file_id}/content\"\n",
+    "        os.makedirs(folder_path, exist_ok=True)\n",
+    "        filename = os.path.join(folder_path, f\"{file_name}\")\n",
+    "        path_list.append(filename)\n",
+    "        headers = {\n",
+    "            \"Authorization\": f\"Bearer {citesphere_token}\",\n",
+    "            \"Content-Type\": \"application/pdf;charset=UTF-8\"\n",
+    "            }\n",
+    "        response = requests.get(giles_url, headers=headers)\n",
+    "        if response.status_code == 200:\n",
+    "            with open(filename, \"wb\") as file:\n",
+    "                file.write(response.content)\n",
+    "    return path_list  \n",
+    "\n",
+    "#Returns the set of tuples containing, file_ids and file_name for downloading\n",
+    "\n",
+    "\"\"\"\n",
+    "this function returns a set of tuples that are pair of filename and their file ids\n",
+    "set : (tuple(file_id,file_name),...)\n",
+    "files to download in a collection item are located in the following fields:\n",
+    "uploadedFile, extractedText, pages, image, text, ocr, additionalFiles\n",
+    "\"\"\"\n",
+    "\n",
+    "def get_set(items_list:list):\n",
+    "    ids=set()\n",
+    "    for values in items_list:\n",
+    "        #getting the file ids in uploadedFile, extractedText, pages, image, text, ocr, additionalFiles\n",
+    "        ids.add((values[\"uploadedFile\"][\"id\"],values[\"uploadedFile\"][\"filename\"]))\n",
+    "        if values[\"extractedText\"] and values[\"extractedText\"]!= \"None\":\n",
```

Comment:
i'm fairly certain, `if["extractedText"]` covers both cases; the only thing you might have to check `if "extractedText" in values`